bigtable/client/rpc_retry_policy_test.cc

```diff
@@ -13,65 +13,90 @@
 // limitations under the License.
 
 #include "bigtable/client/rpc_retry_policy.h"
+#include "bigtable/client/chrono_literals.h"
 
 #include <gtest/gtest.h>
 #include <chrono>
+#include <thread>
 
-// TODO(coryan) - these are useful when writing tests, consider refactoring.
-// TODO(coryan) - these are generally useful, consider submitting to abseil.io
-namespace bigtable {
-namespace chrono_literals {
-constexpr std::chrono::milliseconds operator "" _ms(unsigned long long ms) {
-  return std::chrono::milliseconds(ms);
+namespace {
+/// Refactor two test cases ...
+void check_limited_time(bigtable::RPCRetryPolicy& tested) {
+  using namespace bigtable::chrono_literals;
+  auto start = std::chrono::system_clock::now();
+  for (int i = 0; i != 100; ++i) {
+    auto actual = tested.on_failure(
+        grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"));
+    if (std::chrono::system_clock::now() < start + 50_ms) {
+      EXPECT_TRUE(actual);
+    } else {
+      EXPECT_FALSE(actual);
+      break;
+    }
+    std::this_thread::sleep_for(1_ms);
+  }
+}
+
+/// Create a grpc::Status with a status code for transient errors.
+grpc::Status transient() {
+  return grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again");
+}
+
+/// Create a grpc::Status with a status code for permanent errors.
+grpc::Status permanent() {
```

Comment:
`PermanentError` ?

Though by Google standards, it would probably be `CreatePermanentError()`. ;-)

Reply:
Ditto.